File: Codeforces/Codeforces Round 894 (Div. 3)/A_Gift_Carpet.py
for _ in range(int(input())):
    s = input().split(" ")
    x, y = int(s[0]), int(s[1])
    vika = ["v", "i", "k", "a"]
    s = ""
    for __ in range(x):
        s += input()
    index = int(0)
    for j in range(y):
        for i in range(x):
            if index < 4 and s[y * i + j] == vika[index]:
                index += 1
                break
    if index == 4:
        print("YES")
    else:
        print("NO")

File: Codeforces/Codeforces Round 894 (Div. 3)/A_Gift_Carpet.cpp
# include <iostream>

using namespace std;

string vika = "vika";
char input[20][20];

void solve() {
    int n, m;
    cin >> n >> m;
    for (int i = 0; i < n; i++)
        for (int j = 0; j < m; j++)
            cin >> input[i][j];
    int index = 0;
    for (int j = 0; j < m; j++) {
        for (int i = 0; i < n; i++) {
            if (input[i][j] == vika[index]) {
                index++;
                break;
            }
        }
        if (index >= 4) break;
    }
    if (index == 4) cout << "YES" << endl;
    else cout << "NO" << endl;
    return;
}

int main() {
    int t;
    cin >> t;
    while (t--) solve();
    return 0;
}